src/java.base/share/classes/sun/security/ssl/TransportContext.java

```diff
@@ -584,20 +584,34 @@ private void initiateOutboundClose() throws IOException {
         closeNotify(useUserCanceled);
     }
 
-    // Note; HandshakeStatus.FINISHED status is retrieved in other places.
+    // Note: HandshakeStatus.FINISHED status is retrieved in other places.
     HandshakeStatus getHandshakeStatus() {
         if (!outputRecord.isEmpty()) {
             // If no handshaking, special case to wrap alters or
             // post-handshake messages.
-            return HandshakeStatus.NEED_WRAP;
+            if (!isOutboundClosed()) {
```

Comment:
If I'm reading the [TransportContect#closeNotify](https://github.com/openjdk/jdk/blob/627ad9fe22a153410c14d0b2061bb7dee2c300af/src/java.base/share/classes/sun/security/ssl/TransportContext.java#L275) and [TransportContext#passiveInboundClose](https://github.com/openjdk/jdk/blob/627ad9fe22a153410c14d0b2061bb7dee2c300af/src/java.base/share/classes/sun/security/ssl/TransportContext.java#L524) correctly, non-empty output record with both inbound and outbound closed happens when we reply with our close_notify to peer's. Now we will return NOT_HANDSHAKING which appears to be wrong.

Reply:
Good catch!  I will have an update.  Thank you for the code review.